proto/config.proto

```diff
@@ -0,0 +1,82 @@
+package oci;
+
+//import "config_user.proto";
+
+// Spec is the base configuration for the container.  It specifies platform
+// independent configuration.
+message Spec {
+	// Version is the version of the specification that is supported.
+  optional string Version = 1;
+	// Platform is the host information for OS and Arch.
+  optional Platform Platform = 2; // [default=77];
+	// Process is the container's main process.
+  optional Process Process = 3;
+	// Root is the root information for the container's filesystem.
+  optional Root Root = 4;
+	// Hostname is the container's host name.
+  optional string Hostname = 5;
+	// Mounts profile configuration for adding mounts to the container's filesystem.
+  repeated MountPoint Mounts = 6;
+}
+
+// Platform specifies OS and arch information for the host system that the container
+// is created for.
+message Platform {
+	// OS is the operating system.
+  optional string OS = 1;
+	// Arch is the architecture
+  optional string Arch = 2;
+}
+
+// Process contains information to start a specific application inside the container.
+message Process {
+	// Terminal creates an interactive terminal for the container.
+	optional bool Terminal = 1;
+	// User specifies user information for the process.
+	optional User User = 2;
+	// Args specifies the binary and arguments for the application to execute.
+	repeated string Args = 3;
+	// Env populates the process environment for the process.
+	repeated string Env = 4;
+	// Cwd is the current working directory for the process and must be
+	// relative to the container's root.
+	optional string Cwd = 5;
+}
+
+enum PlatformType {
+  LINUX = 1;
+}
+
+// User specifies user information for the process.
+message User {
+  // Type so that receivers of this message can `switch` for the fields expected
+  optional PlatformType Type = 1 [default = LINUX];
+  // LinuxUserFields are to be used when Type is LINUX
+  optional linuxUserFields LinuxUserFields = 2;
```

Comment:
We can use extensions in proto2.

On Mon, Sep 14, 2015 at 7:41 AM, Vincent Batts <notifications@github.com>
wrote:

> In proto/config.proto
> <https://github.com/opencontainers/specs/pull/179#discussion_r39401570>:
>
> > +	repeated string Env = 4;
> > +	// Cwd is the current working directory for the process and must be
> > +	// relative to the container's root.
> > +	optional string Cwd = 5;
> > +}
> > +
> > +enum PlatformType {
> > +  LINUX = 1;
> > +}
> > +
> > +// User specifies user information for the process.
> > +message User {
> > +  // Type so that receivers of this message can `switch` for the fields expected
> > +  optional PlatformType Type = 1 [default = LINUX];
> > +  // LinuxUserFields are to be used when Type is LINUX
> > +  optional linuxUserFields LinuxUserFields = 2;
>
> @vishh <https://github.com/vishh> ah that sounds like what I'm looking
> for, but that's in proto3. I'm dealing with proto2 :-\
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/specs/pull/179/files#r39401570>.
>